src/modules/presence/presence/presence.gateway.ts

```
@@ -0,0 +1,72 @@
+        .emit('status:update', 'offline');
+    }
+  }
+
+  private async authenticateUser(client: Socket): Promise<User | null> {
+    const authToken = client.handshake.headers.authorization?.split(' ')[1];
+    if (!authToken) return null;
+
+    try {
+      const payload = this.jwtService.verify(authToken);
+      return this.presenceService.getUserById(payload.sub);
+    } catch (error) {
```

ESLint

[error] 'error' is defined but never used.

(@typescript-eslint/no-unused-vars)